Angular/src/app/other_components/expandable-text/expandable-text.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Component, Input, ElementRef, AfterViewInit, ChangeDetectorRef } from '@angular/core';
+//import { CommentaryComponent } from '../../commentary/commentary.component';
+
+@Component({
+  selector: 'app-expandable-text',
+  templateUrl: './expandable-text.component.html',
+  styleUrls: ['./expandable-text.component.scss'],
+})
+export class ExpandableTextComponent implements AfterViewInit {
+  @Input() content: string; // HTML content
+  @Input() isCollapsed?: boolean;
+  @Input() collapsedHeightPx?: number; // Height when collapsed in em;
+
+  protected enabled: boolean;
+  protected maskDisabled = false;
+  protected content_hideable: string;
+
+  constructor(private elRef: ElementRef, private cdRef: ChangeDetectorRef) {
+    this.enabled = true;
+    this.isCollapsed = this.isCollapsed || true; // Set isCollapsed to true by default
+    this.collapsedHeightPx = 200; // Element height is 200px by default when collapsed FIXME: Em would be nicer, but would also make the comparison in the ngAfterViewInit below a bit more difficult.
+    // Send this variable to the css
+    document.querySelector('body').style.setProperty('--collapsed-height', String(this.collapsedHeightPx + 'px'));
+  }
+
+  ngAfterViewInit() {
+    // Check if the content has a hidden section
+    // If so, hide this section, hide the gradient mask and set the component height to fit-content.
+    if (this.content.includes('[hidden]')) {
+      // Hide the to-be-hidden text to the hideable section
+      this.content = this.content.replace('[/hidden]', ''); // Remove leftover tags
+      let content_hideable_arr: Array<string>;
```

Comment:
should we use `string[]` here? It's the same thing, but then it's identical in the entire app (: